optuna/samplers/tpe/sampler.py

```diff
@@ -534,9 +535,15 @@ def _get_observation_pairs(study, param_name):
     if study.direction == StudyDirection.MAXIMIZE:
         sign = -1
 
+    _study = study
+    if isinstance(study.pruner, HyperbandPruner):
```

Comment:
I think that it's helpful for future readers of this repository, to leave a brief comment about why this HyperbandPruner specific logic is necessary.

Reply:
Absolutely!